### PR TITLE
Fully support installPackagePath

### LINKED ISSUE #23515: `istioctl profile` commands do not support --charts

### PR DESCRIPTION
**Pull request by howardjohn**
Currently the profiles are always grabbed from the VFS in some cases.
This is especially broken in development, as the assets.gen.go is not
updated. This changes the last few areas to properly support
installPackagePath. As a result, we no longer check in the profile
to assets.gen.go to github (it is added at release time).

Related: https://github.com/istio/istio/pull/23512
Fixes: https://github.com/istio/istio/issues/23515

### CODE REVIEW

File: operator/pkg/helm/vfs_renderer.go [hunk 1]
```diff
@@ -34,21 +35,22 @@ const (
...
-	profilePaths, err := vfs.ReadDir(profilesRoot)
-	if err != nil {
-		panic(err)
-	}
-	for _, p := range profilePaths {
-		p = strings.TrimSuffix(p, ".yaml")
-		ProfileNames[p] = true
-	}
-}
+//
```
Comment (sdake) [inline operator/pkg/helm/vfs_renderer.go:38]: is this needed?

**Review by sdake — DISMISSED** [review]

**Review by dgn — APPROVED** [review]
lgtm once we remove commented code

File: operator/pkg/helm/vfs_renderer.go (hunk 1 above)
Comment (howardjohn) [reply operator/pkg/helm/vfs_renderer.go]: Removed, can you give T&R review?

**Review by sdake — APPROVED** [review]
LGTM.

**Pull request merged**